sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -42,6 +42,7 @@ import org.apache.spark.sql.execution.{ColumnarRule, SparkPlan}
  * <li>Check Analysis Rules.</li>
  * <li>Optimizer Rules.</li>
  * <li>Pre CBO Rules.</li>
+ * <li>Early Scan Push-Down</li>
```

Comment:
After so many discussions in https://github.com/apache/spark/pull/30808 , I'm really worried about the naming of this new extension point.

In general, this new extension point allows people to inject custom data source operator pushdown rules, which run after the built-in ones. But then the existing `Pre CBO rules` becomes a confusing name, as the pushdown rules are also pre-CBO.

We may need more time to think about the naming, or think if really need custom data source pushdown rules.

Reply:
I feel that the name of `Pre-CBO Rules` is too large and the meaning in the [comment](https://github.com/apache/spark/blob/9596820a473241928b198c1057f4ac662c77dc08/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L212) is not the same. Can't limit the flexibility because of this name.